apis/database/v1beta1/nosql_types.go

```diff
@@ -0,0 +1,164 @@
+	// with a length no greater than 256 characters.
+	// +optional
+	Tags map[string]string `json:"tags,omitempty"`
+}
+
+// CosmosDBAccountObservation shows current state of an Azure CosmosDB account.
+type CosmosDBAccountObservation struct {
+	// Identity - The identity of the resource.
+	ID string `json:"id"`
+
+	// State - current state of the account in Azure.
+	State string `json:"state"`
+
+	// Kind - Indicates the type of database account.
+	Kind documentdb.DatabaseAccountKind `json:"kind"`
+
+	// Location - The location of the resource. This will be one of the
+	// supported and registered Azure Geo Regions (e.g. West US, East US,
+	// Southeast Asia, etc.).
+	Location string `json:"location"`
+
+	// Properties - Account properties like databaseAccountOfferType, ipRangeFilters, etc.
+	Properties CosmosDBAccountProperties `json:"properties"`
+
+	// Tags - A list of key value pairs that describe the resource. These tags
+	// can be used for viewing and grouping this resource (across resource
+	// groups). A maximum of 15 tags can be provided for a resource. Each tag
+	// must have a key with a length no greater than 128 characters and a value
+	// with a length no greater than 256 characters.
+	// +optional
+	Tags map[string]string `json:"tags,omitempty"`
+}
+
+// CosmosDBAccountProperties define the desired properties of an Azure CosmosDB account.
+type CosmosDBAccountProperties struct {
+	// ConsistencyPolicy - The consistency policy for the Cosmos DB account.
+	ConsistencyPolicy *CosmosDBAccountConsistencyPolicy `json:"consistencyPolicy,omitempty"`
+	// Locations - An array that contains the georeplication locations enabled for the Cosmos DB account.
+	Locations []CosmosDBAccountLocation `json:"locations"`
+	// DatabaseAccountOfferType - The offer type for the database
+	DatabaseAccountOfferType string `json:"databaseAccountOfferType"`
+	// IPRangeFilter - Cosmos DB Firewall Support: This value specifies the set of IP addresses or IP address ranges in CIDR form to be included as the allowed list of client IPs for a given database account. IP addresses/ranges must be comma separated and must not contain any spaces.
+	IPRangeFilter *string `json:"ipRangeFilter,omitempty"`
+	// EnableAutomaticFailover - Enables automatic failover of the write region in the rare event that the region is unavailable due to an outage. Automatic failover will result in a new write region for the account and is chosen based on the failover priorities configured for the account.
+	EnableAutomaticFailover *bool `json:"enableAutomaticFailover,omitempty"`
+	// EnableMultipleWriteLocations - Enables the account to write in multiple locations
+	EnableMultipleWriteLocations *bool `json:"enableMultipleWriteLocations,omitempty"`
+	// EnableCassandraConnector - Enables the cassandra connector on the Cosmos DB C* account
+	EnableCassandraConnector *bool `json:"enableCassandraConnector,omitempty"`
+}
```

Comment:
Nit: Some of these fields are pointers with `omitempty`, which makes me think they're optional. It would be good to add the `// + optional` tag to indicate that.